go/client/tcp_conn.go

```diff
@@ -168,6 +169,8 @@ func (conn *tcpConn) OnPacket(fn func(*protocol.Packet, error)) {
 }
 
 func (conn *tcpConn) Close(err error) {
+	conn.mu.Lock()
```

Comment:
I consider that sync.Once is better for this situation. What do you think?

Reply:
I agree.